IdentityCore/src/webview/systemWebview/ios/MSIDAuthenticationSession.m

```diff
@@ -139,14 +193,22 @@ - (void)startWithCompletionHandler:(MSIDWebUICompletionHandler)completionHandler
             _webAuthSession = [[ASWebAuthenticationSession alloc] initWithURL:_startURL
                                                             callbackURLScheme:_callbackURLScheme
                                                             completionHandler:authCompletion];
+            if (@available(iOS 13.0, *))
+            {
+                _webAuthSession.presentationContextProvider = self;
+                _webAuthSession.prefersEphemeralWebBrowserSession = self.prefersEphemeralWebBrowserSession;
+            }
+            
             if ([_webAuthSession start]) return;
         }
         else
         {
+#if !TARGET_OS_UIKITFORMAC
```

Comment:
nit: it's becoming a bit difficult to read with all  the @available and #if-s (thanks to all the new APIs). Would it make sense to modularize and have the same interface for all of them that returns session and performs actions on the session? And a factory that creates a session based on current iOS version, macOS frameworks etc?